vllm/engine/llm_engine.py

```diff
@@ -259,7 +260,9 @@ def __init__(
             prompt_adapter_config=prompt_adapter_config,
         )
 
-        if not self.model_config.embedding_mode:
+        if self.model_config.model_mode not in [
+                ModelMode.EMBEDDING, ModelMode.SIMPLE
+        ]:
```

Comment:
Perhaps we should let each mode declare whether to set up KV cache instead of handling each case inside `LLMEngine`.